src/Components/Endpoints/src/RazorComponentEndpointInvoker.cs

```diff
@@ -145,8 +149,17 @@ private async Task<RequestValidationState> ValidateRequestAsync(IAntiforgery? an
     private string? GetFormHandler(out bool isBadRequest)
     {
         isBadRequest = false;
+        var hasValue = false;
+        StringValues value = StringValues.Empty;
+        try
+        {
+            hasValue = _context.Request.Form.TryGetValue("_handler", out value);
+        }
+        // We may be trying to read a Form that's been poisoned because
+        // the anti-forgery token was not valid.
+        catch (InvalidOperationException) { }
```

Comment:
We probably don't need to check for `InvalidOperationException` here. Since we always check/validate antiforgery, we can probably return after https://github.com/dotnet/aspnetcore/pull/49233/files#diff-b891c4aa1da07f8709ffc51ce6c26c489d9432866aee1cdac187484377dbfefeL132-R139